lib/r10k/environment/base.rb

```diff
@@ -43,7 +43,7 @@ def initialize(name, basedir, dirname, options = {})
     @basedir = basedir
     @dirname = dirname
     @options = options
-    @puppetfile_name = options[:puppetfile_name]
+    @puppetfile_name = @options.delete(:puppetfile_name)
```

Comment:
Ooh, yes, and drawing attention to this caught another problem, 'cause the derived classes all use `options` now instead of `@options` ... will add a fix for this.